README
# UNMAINTAINED
Please note this project is unmaintained. All updates seen here are just out
of pure coincidence.

JWChat - Jabber Web Chat
========================

JWChat aims to be a full featured, web based jabber client. It uses
only JavaScript and HTML on the client-side. Currently it supports
basic jabber instant messaging, roster management and muc-based
groupchats.

If you want to learn more about Jabber please refer to
http://www.jabber.org.

JWChat is a web based instant messenger (IM) just like AIM, MSN
Messenger, Yahoo! Messenger or ICQ. This means you can manage your
contacts and chat with other users directly. Unlike other IMs you can
use this with your web browser without having to install any
additional software at all. For a list of supported browsers see
below. Your contact list is stored on the server. This means that you
have access to your contact list from almost any computer as long as
you've got access to the internet.

Supported browsers and plattforms
=================================

Windows XP/2000/ME/98/95:
* MS Internet Explorer version >= 5.0
* Netscape version 6 & 7
* Mozilla (stable) version >= 1.0.2
* Firefox 2.x and 3.x
* Opera since v9.x
  
Mac OS X:
* Mozilla (stable) version >= 1.0.2
* Firefox 2.x and 3.x
* Camino 0.7
* Safari since version 3

Linux:
* Mozilla since version >= 1.0.2
* Firefox 2.x and 3.x
* Konqueror since v?
* Opera since 9.x


Requirements
============

You need to have access to a jabber server by way of either [1]HTTP Polling
or [2]HTTP Binding. Either use some server that has support for one of these
protocols biult in (e.g. [3]ejabberd) or use some gateway like [4]punjab 
(python based) or [5]JabberHTTPBind (a java servlet).

[1]http://www.jabber.org/jeps/jep-0025.html
[2]http://www.jabber.org/jeps/jep-0124.html
[3]http://ejabberd.jabber.ru
[4]http://punjab.sourceforge.net
[5]http://zeank.in-berlin.de/jhb/

Additionally you need a web-server of your choice capable
of doing address rewriting.

Download
========

Please refer to http://jwchat.sourceforge.net/download.shtml for
download instructions.

Quick Install Guide
===================

* If you attained JWChat by checking it out via CVS, be sure to also
  read README.CVS in the top-level directory.

* Unpack jwchat into some directory that is accessible by your web
  server. E.g.: /var/www/jwchat

* Setup your web server so that it redirects some local address like
  http://jabber.example.com/http-poll/ to some service that either
  implements HTTP Polling or HTTP Binding.

  Some servers like ejabberd do have such a service built-in. Please
  refer to the docs of your jabber server to find out how to use it.

  If your server does not have such a service built-in or you don't
  plan to run your own jabber server at all there are standalone
  components like punjab or JabberHTTPBind available that let you
  connect to any jabber server using HTTP Binding or HTTP Polling.

  Here is an example that should work with a default ejabberd
  installation on the same host:

%<-----<schnipp>-----
<VirtualHost *>
  ServerName jabber.example.com
  DocumentRoot /var/www/jwchat
  <Directory /var/www/jwchat>
    Options  +Indexes +MultiViews
  </Directory>
  AddDefaultCharset UTF-8
  RewriteEngine on
  RewriteRule http-poll/ http://127.0.0.1:5280/http-poll/ [P]
</VirtualHost>
%<-----<schnapp>-----

  Alternatively you could put a .htaccess file within jwchat's
  subdirectory like this (you need to have 'AllowOverride All' in your
  apache config in order to make this work):

%<-----<schnipp>-----
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteRule http-poll/ http://127.0.0.1:5280/http-poll/ [P]
</IfModule>
%<-----<schnapp>-----

  Note 1: You need to enable mod_rewrite and mod_proxy at your apache
  configuration. Apache2 users also need to enable the protocol
  specific proxy module mod_proxy_http.

  Note 2: With JabberHTTPBind you don't need to do this sort of
  address rewriting if you're serving JWChat's files from within your
  servlet container directly or you're using some technique like
  mod_jk to mount a remote servlet engine directory transparently into
  your apache's domain of served addresses.

  Note to FreeBSD users: Please add 'WITH_PROXY_MODULES=true' to your 
  /etc/make.conf before installing apache!

* Restart your web server.

* Test if this redirect works. Following the example above point your
  browser to http://jabber.example.com/http-poll/. You should see the
  same response as if you you're pointing it to
  http://jabber.example.com:5280/http-poll/.

* Edit 'config.js' to suit your needs.
  Be sure to set 'httpbase' to the redirected address from
  above. Within our example it would have to be
  'http://jabber.example.com/http-poll/' or 'http-poll/' in short.
  
  Note 1: Double check this if using some relative address (relative to
  jwchat's base URL).

  Note 2: You can't use an httpbase address that is not local to your
  JWChat installation. Local in this context means that it MAY NOT be
  on some different host or port then where JWChat is being served from.

* If you haven't done yet start your jabber server (if any).

* Point your browser to http://jabber.example.com and log in.

Disclaimer
==========

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or (at
your option) any later version.

This program is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
USA


ChangeLog
2005-12-06  Stefan Strigler  <steve@zeank.in-berlin.de>

	* src/roster.html: fixed style attrib 'cursor'

	* src/options.html: fixed style tag comments 

2005-12-05  Stefan Strigler  <steve@zeank.in-berlin.de>

	* src/index.html:
	* src/jwchat.html:
	- support presence priorities
	- store onlmsg at server side, load on startup
	* src/jwchat.css
	- use 'cursor: pointer;' throughout
	* src/chat_iframe.html:
	* src/groupchat_iframe.html:
	* src/help.html: fixed style tag comments

2005-12-01  Stefan Strigler  <steve@zeank.in-berlin.de>

	* po/sk.po: Slovak translation added

2005-08-25  Stefan Strigler  <steve@zeank.in-berlin.de>

	* po/pt_br.po: added

2005-07-21  Stefan Strigler  <steve@zeank.in-berlin.de>

	* README: be more verbose

2005-06-24  Stefan Strigler  <steve@zeank.in-berlin.de>

	* po/nl.po: update translation for nl

2004-10-25  Stefan Strigler  <steve@zeank.in-berlin.de>

	* use JSJaC for communication. support for WebClientService has
	been dropped in favor of HTTP-Polling/HTTP-Binding

2004-09-21  Stefan Strigler  <steve@zeank.in-berlin.de>

	* support for making annotations to users

2004-09-20  Stefan Strigler  <steve@zeank.in-berlin.de>

	* added about tab to options dialog

2004-09-14  Stefan Strigler  <steve@zeank.in-berlin.de>

	* added support for digest login

2004-09-09  Stefan Strigler <steve@zeank.in-berlin.de>

	* did a lot of bug fixing the last days

2004-08-17  Stefan Strigler <steve@zeank.in-berlin.de>

	* initial support for JEP-0136 Message Archiving

2004-07-09  Stefan Strigler <steve@zeank.in-berlin.de>

	* fixed user info - fetching vcard works now
	* implemented irc-style commands for muc
	* fixed timestamps for messages

2004-07-08  Stefan Strigler <steve@zeank.in-berlin.de>

	* support for muc invite
	* muc room names are escaped now
	* redesign chat dialog

2004-07-02  Stefan Strigler <steve@zeank.in-berlin.de>

	* load/store roster preferences (hidden/collapsed groups)
	
2004-07-01  Stefan Strigler <steve@zeank.in-berlin.de>

	* use Private XML Storage (JEP-0049) to store client preferences

2004-06-28  Stefan Strigler <steve@zeank.in-berlin.de>

	* support for configuring chatrooms
	* modify user lists (ban, moderators, members and so on)

2004-06-19  Stefan Strigler <steve@zeank.in-berlin.de>

	* support for JUD

2004-06-04  Stefan Strigler <steve@zeank.in-berlin.de>

	* initial support for gateways aka transports
	* started to write a ChangeLog ;-)


src/config.js
/*
 * JWChat, a web based jabber client

 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 *
 * Please visit http://jwchat.sourceforge.net for more information!
 */

/*
 * This is the main configuration file for the chat client itself.
 * You have to edit this before you can use jwchat on your website!
 *
 * Have a look at the README for hints and troubleshooting!
 */

var SITENAME = "jwchat.org";

/* BACKENDS
 * Array of objects each describing a backend.
 *
 * Required object fields:
 * name      - human readable short identifier for this backend
 * httpbase  - base address of http service [see README for details]
 * type      - type of backend, must be 'polling' or 'binding'
 *
 * Optional object fields:
 * description     - a human readable description for this backend
 * servers_allowed - array of jabber server addresses users can connect to 
 *                   using this backend
 *
 * If BACKENDS contains more than one entry users may choose from a
 * select box which one to use when logging in.
 *
 * If 'servers_allowed' is empty or omitted user is presented an input
 * field to enter the jabber server to connect to by hand.
 * If 'servers_allowed' contains more than one element user is
 * presented a select box to choose a jabber server to connect to.
 * If 'servers_allowed' contains one single element no option is
 * presented to user.
 */
var BACKENDS = 
[
		{
			name:"node-xmpp-bosh",
			description:"An XMPP BOSH & WebSocket server (connection manager) written using node.js in Javascript",
			httpbase:"/http-bind/",
			type:"binding",
			default_server: "jwchat.org"
    }/*,
		{
			name:"JabberHTTPBind",
			description:"A Java&trade; servlet implementing JEP-0124 (<a href='http://zeank.in-berlin.de/jhb/' target='_new'>http://zeank.in-berlin.de/jhb/</a>)",
			httpbase:"/JHB/",
			type:"binding",
			default_server: "jwchat.org"
    }/*,
    {
      name:"Ejabberd's HTTP Polling",
      description:"HTTP Polling backend of local ejabberd service",
      httpbase:"/http-poll/",
      type:"polling",
			servers_allowed: ['jwchat.org', 'x-berg.de', 'muckl.org'],
      default_server:"jwchat.org"
    }*/
];

var DEFAULTRESOURCE = "jwchat";
var DEFAULTPRIORITY = "10";

/* DEFAULTCONFERENCEGROUP + DEFAULTCONFERENCESERVER
 * default values for joingroupchat form
 */
var DEFAULTCONFERENCEROOM = "jwchat";
var DEFAULTCONFERENCESERVER = "conference.jwchat.org";

/* debugging options */
var DEBUG = true; // turn debugging on
var DEBUG_LVL = 4; // debug-level 0..4 (4 = very noisy)

var USE_DEBUGJID = true; // if true only DEBUGJID gets the debugger
var DEBUGJID = "zeank2@jwchat.org"; // which user get's debug messages

// most probably you don't want to change anything below

var timerval = 2000; // poll frequency in msec

var stylesheet = "jwchat.css";
var THEMESDIR = "themes";
